components/org.wso2.carbon.identity.password.history/src/test/java/org/wso2/carbon/identity/password/history/handler/PasswordHistoryValidationHandlerTest.java

```diff
@@ -0,0 +1,124 @@
+        Assert.assertFalse(Boolean.parseBoolean(passwordHistoryValidationHandler.getName()), "Invalid getName().");
+
+    }
+
+    @Test
+    public void testGetFriendlyName() throws Exception {
+        Assert.assertEquals(passwordHistoryValidationHandler.getFriendlyName(), "Password History", "Valid getFriendlyName().");
+        Assert.assertNotEquals(passwordHistoryValidationHandler.getFriendlyName(), "passwordHistory", "Invalid getFriendlyName().");
+    }
+
+    @Test
+    public void testGetCategory() throws Exception {
+        Assert.assertEquals(passwordHistoryValidationHandler.getCategory(), "Password Policies", "Valid getCategory().");
+        Assert.assertNotEquals(passwordHistoryValidationHandler.getCategory(), "PasswordPolicies", "Invalid getCategory().");
+    }
+
+
+    @Test
+    public void testGetSubCategory() throws Exception {
+        Assert.assertEquals(passwordHistoryValidationHandler.getSubCategory(), "DEFAULT", "Valid getSubCategory().");
+        Assert.assertNotEquals(passwordHistoryValidationHandler.getSubCategory(), "PasswordPolicies", "Invalid getSubCategory().");
+    }
+
+    @Test
+    public void testGetOrder() throws Exception {
+        Assert.assertEquals(passwordHistoryValidationHandler.getOrder(), 0, "Valid getOrder().");
+        Assert.assertNotEquals(passwordHistoryValidationHandler.getOrder(), 2, "Invalid getOrder().");
+    }
+
+    @Test
+    public void testGetPropertyNameMapping() throws Exception {
+        Map<String, String> expectedMappingTrue = new HashMap<>();
+        expectedMappingTrue.put(PasswordHistoryConstants.PW_HISTORY_ENABLE, "Enable Password History Feature");
+        expectedMappingTrue.put(PasswordHistoryConstants.PW_HISTORY_COUNT, "Password History validation count");
+        Assert.assertEquals(passwordHistoryValidationHandler.getPropertyNameMapping(), expectedMappingTrue, "Valid getPropertyNameMapping() .");
+
+        Map<String, String> expectedMappingFalse = new HashMap<>();
+        expectedMappingFalse.put(PasswordHistoryConstants.PW_HISTORY_ENABLE, "Enable Password History");
+        expectedMappingFalse.put(PasswordHistoryConstants.PW_HISTORY_COUNT, "Password History validation");
+        Assert.assertNotEquals(passwordHistoryValidationHandler.getPropertyNameMapping(), expectedMappingFalse, "Invalid getPropertyNameMapping() .");
+    }
+
+    @Test
+    public void testGetPropertyDescriptionMapping() throws Exception {
+        Map<String, String> expecteddescriptionMappingTrue = new HashMap<>();
+        expecteddescriptionMappingTrue.put(PasswordHistoryConstants.PW_HISTORY_ENABLE, "Enable to disallow previously used passwords");
+        expecteddescriptionMappingTrue.put(PasswordHistoryConstants.PW_HISTORY_COUNT, "Restrict reusing last x number of password during password update");
+        Assert.assertEquals(passwordHistoryValidationHandler.getPropertyDescriptionMapping(), expecteddescriptionMappingTrue, "Valid getPropertyDescriptionMapping().");
```

Comment:
some lines seems to be long.
Stick to 120 length, in compliance with wso2 template.

Reply:
Done